File: targets/python-web/vulns/vuln24_redirect.py
from flask import Blueprint, request, redirect
bp = Blueprint("vuln24", __name__)

@bp.get("/vuln24/redirect")
def r():
    url = request.args.get("url", "")
    # VULN:VULN-24:CWE-601:taint open redirect — unvalidated redirect target
    return redirect(url)

File: targets/jsts-web/src/vulns/vuln24_redirect.ts
import { Router } from "express";
export const router = Router();
router.get("/vuln24/redirect", (req, res) => {
  // VULN:VULN-24:CWE-601:taint open redirect — unvalidated redirect target
  res.redirect(String(req.query.url || ""));
});
